src/App.tsx

```diff
@@ -1,36 +1,55 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState(moviesFromServer);
+
+  const filterMovies = (searchQuery: string) => {
+    const filteredMovies = moviesFromServer.filter(
+      (movie) => movie.title.toLowerCase()
+        .includes(searchQuery.trim().toLowerCase())
+        || movie.description.toLowerCase()
+          .includes(searchQuery.trim().toLowerCase()),
+    );
+
+    setVisibleMovies(filteredMovies);
+  };
```

Comment:
it can be with `useCallback`, whick is dependednt on moviesFromServer and retirns the filtered movies instead of set them with setVisibleMovies
```suggestion
  const filterMovies = useCallback((searchQuery: string) => {
    return moviesFromServer.filter(
      (movie) => movie.title.toLowerCase()
        .includes(searchQuery.trim().toLowerCase())
        || movie.description.toLowerCase()
          .includes(searchQuery.trim().toLowerCase()),
    );
  }, [moviesFromServer]);
```